src/module/address/address.module.ts

```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Address } from './domain/address.entity';
+import { AddressController } from './address.controller';
+import { AddressService } from './application/service/address.service';
+import { AddressRepository } from './infrastructure/address.repository';
+
+@Module({
+    imports: [TypeOrmModule.forFeature([Address])],
+    controllers: [AddressController],
+    providers: [AddressService, AddressRepository],
```

Comment:
We should be using the  { provide: ADDRESS_REPOSITORY,  useClass: AddressRepository }
where ADDRESS_REPOSITORY is a const that has the value 'ADDRESS_REPOSITORY'. This const should be in the same file as the repository interface.  
If we do it this way, then changing repositories is as simple as changing the useClass properties. 
When you use the @Inject() decorator, you should be using  @Inject(ADDRESS_REPOSITORY).

Reply:
done